src/esaml_sp.erl

```diff
@@ -281,6 +291,46 @@ validate_assertion(Xml, DuplicateFun, SP = #esaml_sp{}) ->
     ], Xml).
 
 
+%% @doc Decrypts an encrypted assertion element.
+-spec decrypt_assertion(xml(), esaml:sp()) ->
```

Comment:
This function returns `#xmlElement{}` on success. The spec doesn't match that. May be just remove this and dialyzer do its thing automatically?

Reply:
Ok, I've removed the spec.